extends StaticBody2D

@onready var area = $Area2D

var dialog_checkpoint = [
	{
		"text": "쮻ese치s guardar la sesi칩n?",
		"options": [
			{"text": "S칤", "next": -1},
			{"text": "No", "next": -1}
		]
	},
	"Juego guardado."
]

func _ready():
	area.input_event.connect(_on_area_input)

func _on_area_input(_viewport, event, _shape_idx):
	if event is InputEventMouseButton and event.pressed and event.button_index == MOUSE_BUTTON_LEFT:
		GESTORDIALOGOS.show_external_dialog(dialog_checkpoint.duplicate(true))
		GESTORDIALOGOS.dialog_completed_callback = save_after_confirmation

func save_after_confirmation(index: int):
	# Si el jugador eligi칩 la primera opci칩n (칤ndice 0), guarda.
	print("游릭 save_after_confirmation ejecutado con 칤ndice:", index)
	if index == 0:
		GAMESTATE.save_game()
		GESTORDIALOGOS.show_external_dialog(["Juego guardado."])
